astropy/table/table.py

```diff
@@ -1618,12 +1662,58 @@ def add_columns(self, cols, indexes=None, copy=True, rename_duplicate=False):
               2 0.2 1.2  y
               3 0.3 1.3  z
 
+        Add unnamed columns or mixin objects at the end of the table using default names::
+            >>> t = Table([[1, 2, 3], [0.1, 0.2, 0.3]], names=('a', 'b'))
+            >>> col_c = Column(data=['x', 'y', 'z'])
+            >>> col_d = Column(data=['u', 'v', 'w'])
+            >>> t.add_columns([col_c, col_d])
+            >>> print(t)
+             a   b   2   3
+            --- --- --- ---
+              1 0.1   x   u
+              2 0.2   y   v
+              3 0.3   z   w
+
+        Add unnamed columns or mixin objects at the end of the table by specifying explicit names::
+            >>> t = Table([[1, 2, 3], [0.1, 0.2, 0.3]], names=('a', 'b'))
+            >>> col_c = Column(data=['x', 'y', 'z'])
+            >>> col_d = Column(data=['u', 'v', 'w'])
+            >>> t.add_columns([col_c, col_d], names=['c', 'd'])
+            >>> print(t)
+             a   b   c   d
+            --- --- --- ---
+              1 0.1   x   u
+              2 0.2   y   v
+              3 0.3   z   w
+
+        Add columns or mixin objects at the end of the table and specify different names for each::
+            >>> t = Table([[1, 2, 3], [0.1, 0.2, 0.3]], names=('a', 'b'))
+            >>> col_b = Column(name='b', data=[1.1, 1.2, 1.3])
+            >>> col_c = Column(name='c', data=['x', 'y', 'z'])
+            >>> t.add_columns([col_b, col_c], names=['c', 'd'])
+            >>> print(t)
+             a   b   c   d
+            --- --- --- ---
+              1 0.1 1.1   x
+              2 0.2 1.2   y
+              3 0.3 1.3   z
         """
         if indexes is None:
             indexes = [len(self.columns)] * len(cols)
         elif len(indexes) != len(cols):
             raise ValueError('Number of indexes must match number of cols')
 
+        if names is None:
+            names = [str(name) for name in range(len(self.columns), len(self.columns) + len(cols))]
+            for col, name in zip(cols, names):
+                if col.info.name is None:
+                    col.info.name = name
```

Comment:
I don't think we want to set the names of the input columns -- that may not be expected by the user; instead, use `names` much further down, in the construction of the list of new columns to initialize the table with.